app/services/data_migrations/fix_candidate_api_updated_at.rb

```diff
@@ -0,0 +1,48 @@
+module DataMigrations
+  class FixCandidateAPIUpdatedAt
+    TIMESTAMP = 20211208172459
+    MANUAL_RUN = false
+
+    # Ensure that all the candidates have a `candidate_api_updated_at` that is
+    # no earlier than the `created_at` of the most recent application form.
+    def change
+      candidates = Candidate
+        .joins(
+          'inner join (
+            select candidates.id id,
+              max(application_forms.created_at) max_form_created_at
+            from candidates
+            inner join application_forms on application_forms.candidate_id = candidates.id
+            group by candidates.id
+          ) candidates_with_latest_application_form on candidates_with_latest_application_form.id = candidates.id',
+        )
+        .joins(
+          "left outer join (
+            select application_forms.candidate_id candidate_id,
+              min(audits.created_at) earliest_update
+            from audits
+            inner join application_forms on application_forms.id = audits.auditable_id
+            WHERE audits.action = 'update'
+            AND audits.auditable_type = 'ApplicationForm'
+            AND audits.user_id IS NOT NULL
+            group by application_forms.candidate_id
```

Comment:
@duncanjbrown I tweaked this join a bit following our call about it. We need to group by `candidate.id` by itself (not include `application_forms.id`) so that we work out the minimum values per candidate rather per application. Doing the later leads to multiple values of the 'minimum' meaning that you sometimes get the right answer (confusingly). I've also simplified the join condition a bit so that both joins are independent (because the minimum update time and maximum create times occur on different application forms for a given candidate).